File: python-sdk/src/ace_sdk/_internal/deployment.py
# Copyright (c) Aptos Labs
# SPDX-License-Identifier: Apache-2.0
"""Mirrors src/_internal/deployment.ts."""

from __future__ import annotations

from typing import Any

from aptos_sdk.account_address import AccountAddress


class AceDeployment:
    def __init__(
        self,
        api_endpoint: str,
        contract_addr: AccountAddress,
        api_key: str | None = None,
        client_config: dict[str, Any] | None = None,
        discovery_url: str | None = None,
    ) -> None:
        self.api_endpoint = api_endpoint
        self.contract_addr = contract_addr
        self.api_key = api_key
        self.client_config = client_config
        self.discovery_url = discovery_url

    def with_api_key(self, api_key: str | None = None) -> "AceDeployment":
        return AceDeployment(
            api_endpoint=self.api_endpoint,
            contract_addr=self.contract_addr,
            api_key=api_key,
            client_config=self.client_config,
            discovery_url=self.discovery_url,
        )

    def with_client_config(
        self, client_config: dict[str, Any] | None = None
    ) -> "AceDeployment":
        return AceDeployment(
            api_endpoint=self.api_endpoint,
            contract_addr=self.contract_addr,
            api_key=self.api_key,
            client_config=client_config,
            discovery_url=self.discovery_url,
        )

File: rust-sdk/src/aptos/deployment.rs
// Copyright (c) Aptos Labs
// SPDX-License-Identifier: Apache-2.0

//! Mirrors `ts-sdk/src/_internal/deployment.ts`.

use crate::address::AccountAddress;

/// Where an ACE deployment lives: a fullnode REST endpoint (`.../v1`) and the ACE contract
/// address, optionally an API key (sent as `Authorization: Bearer`) and a discovery service URL.
#[derive(Clone, PartialEq, Eq, Debug)]
pub struct AceDeployment {
    pub api_endpoint: String,
    pub contract_addr: AccountAddress,
    pub api_key: Option<String>,
    pub discovery_url: Option<String>,
}

impl AceDeployment {
    pub fn new(api_endpoint: impl Into<String>, contract_addr: AccountAddress) -> Self {
        Self {
            api_endpoint: api_endpoint.into(),
            contract_addr,
            api_key: None,
            discovery_url: None,
        }
    }
    pub fn with_api_key(mut self, api_key: Option<String>) -> Self {
        self.api_key = api_key;
        self
    }
    pub fn with_discovery_url(mut self, discovery_url: Option<String>) -> Self {
        self.discovery_url = discovery_url;
        self
    }
}
